### PR TITLE
Datasource: Add support for max_conns_per_host in dataproxy settings

### PR DESCRIPTION
**Pull request by jvrplmlmn**
**What this PR does / why we need it**: This PR brings to main the ability to configure the `MaxConnsPerHost` _knob_ , in a similar fashion to https://github.com/grafana/grafana/pull/35519. 

**Which issue(s) this PR fixes**: Not fixing it, but related to https://github.com/grafana/grafana/issues/35257

**Special notes for your reviewer**: Not sure if contributing this to `main` will backport it to `v8.0.x` or if I should backport that manually.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -51,7 +51,7 @@ require (
 	github.com/gorilla/websocket v1.4.2
 	github.com/gosimple/slug v1.9.0
 	github.com/grafana/grafana-aws-sdk v0.4.0
-	github.com/grafana/grafana-plugin-sdk-go v0.104.0
+	github.com/grafana/grafana-plugin-sdk-go v0.104.1-0.20210610153358-11c9baf0fd86
```
Comment (jvrplmlmn) [inline go.mod:54]: I'm temporarily vendoring https://github.com/grafana/grafana-plugin-sdk-go/pull/370 to make the CI build for this PR to pass. Once that is merged, I'll update the vendor with the right version.

File: docs/sources/administration/configuration.md [hunk 2]
```diff
@@ -414,6 +414,11 @@ The length of time that Grafana will wait for a successful TLS handshake with th
 
 The length of time that Grafana will wait for a datasource’s first response headers after fully writing the request headers, if the request has an “Expect: 100-continue” header. A value of `0` will result in the body being sent immediately. Default is `1` second. For more details check the [Transport.ExpectContinueTimeout](https://golang.org/pkg/net/http/#Transport.ExpectContinueTimeout) documentation.
 
+### max_conns_per_host
+
+Optionally limits the total number of connections per host, including connections in the dialing, active, and idle states. On limit violation, dials are blocked. A value of `0` means that there are no limits. Default is `0`.
```
Comment (jvrplmlmn) [inline docs/sources/administration/configuration.md:419]: For reviewers: Same docs that were added in https://github.com/grafana/grafana/pull/35519

File: go.mod (hunk 1 above)
Comment (marefr) [reply go.mod:54]: Sure that works. We normally recommend to use a replace directive instead when working towards branches, e.g. `go mod edit -replace=github.com/grafana/grafana-plugin-sdk-go=github.com/grafana/grafana-plugin-sdk-go@<branch commit>`

Comment (jvrplmlmn) [reply go.mod]: This was addressed in https://github.com/grafana/grafana/pull/35520/commits/782e874c5940d060da11bf790d50f95de3c4b096

**Review by marefr — APPROVED** [review]
LGTM great

**Pull request merged**